adlfs/spec.py

```diff
@@ -69,7 +69,7 @@
     "is_current_version",
 ]
 _ROOT_PATH = "/"
-_DEFAULT_BLOCK_SIZE = 4 * 1024 * 1024
+_DEFAULT_BLOCK_SIZE = 50 * 2**20
```

Comment:
We should also make sure to update the changelog in this PR. I'm thinking we can just use the three bullet points from this comment I made: https://github.com/fsspec/adlfs/pull/509#issuecomment-3180475337 and make the wording a bit more succinct.